### PR TITLE
Move/Rename OrbitCore::LinuxUtils into OrbitService::Utils

### PR DESCRIPTION
**Pull request by antonrohr**
This moves LinuxUtils out of OrbitCore into OrbitService in 3 separate commits and does some very minor refactoring. 
I know that the new OrbitService::Utils class needs some bigger refactoring and also unit tests. I refrained from doing that in this PR because my current goal is loosing OrbitCore as a dependency of OrbitService. 
After this PR the only thing left that is used in Service and Client is SymbolHelper.

### CODE REVIEW

File: OrbitService/LinuxUtils.cpp [hunk 1]
```diff
@@ -166,4 +176,24 @@ ErrorMessageOr<std::string> GetExecutablePath(int32_t pid) {
...
+outcome::result<std::string> FileToString(
+    const std::filesystem::path& file_name) {
+  std::ifstream file_stream(file_name);
+  if (file_stream.fail()) {
+    return outcome::failure(static_cast<std::errc>(errno));
+  }
+  return outcome::success(
+      std::string{std::istreambuf_iterator<char>{file_stream},
+                  std::istreambuf_iterator<char>{}});
+}
```
Comment (dimitry-) [inline OrbitService/LinuxUtils.cpp]: please add tests for this and other methods in this file.

Comment (dimitry-) [reply OrbitService/LinuxUtils.cpp]: I also wonder if this should be moved to base - seems pretty generic.

File: OrbitService/LinuxUtils.cpp [hunk 2]
```diff
@@ -166,4 +176,24 @@ ErrorMessageOr<std::string> GetExecutablePath(int32_t pid) {
   return std::string(buffer, length);
 }
 
+outcome::result<std::string> FileToString(
```
Comment (dimitry-) [inline OrbitService/LinuxUtils.cpp]: ErrorMessageOr?

File: OrbitService/LinuxUtils.cpp (hunk 1 above)
Comment (antonrohr) [reply OrbitService/LinuxUtils.cpp]: Like I said in the PR description, I wanted to skip refactoring and tests in this PR. But I hear you and I can them in this PR. I will add a 4th commit for that

File: OrbitService/LinuxUtils.cpp (hunk 2 above)
Comment (antonrohr) [reply OrbitService/LinuxUtils.cpp]: I change that in the 3rd of these commits. I could move it to commit 1, but I dont think it actually matters?

Comment (dimitry-) [reply OrbitService/LinuxUtils.cpp]: it's fine as it is.

**Comment by antonrohr** [comment]
I added tests for OrbitService::Utils. ptal

Because of the clang-format change I had to force push, just fyi

File: OrbitService/Utils.cpp [hunk 3]
```diff
@@ -121,32 +130,43 @@ ErrorMessageOr<std::vector<ModuleInfo>> ListModules(int32_t pid) {
...
 }
 
-outcome::result<bool> Is64Bit(pid_t pid) {
+ErrorMessageOr<bool> Is64Bit(pid_t pid) {
   // TODO(161196904): Do this in a more reliable way. It does not work for a lot
   //  of processes.
-  OUTCOME_TRY(result, ExecuteCommand(absl::StrFormat("file -L /proc/%d/exe", pid)));
-  return absl::StrContains(result, "64-bit");
+  OUTCOME_TRY(file_data, ExecuteCommand(absl::StrFormat("file -L /proc/%d/exe", pid)));
+  return Is64Bit(file_data);
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:162]: maybe it is time to use ElfFile here.

File: OrbitService/Utils.cpp [hunk 4]
```diff
@@ -155,7 +175,26 @@ ErrorMessageOr<std::string> GetExecutablePath(int32_t pid) {
                                         pid, SafeStrerror(errno)));
   }
 
-  return std::string(buffer, length);
+  return Path{std::string(buffer, length)};
+}
+
+ErrorMessageOr<std::string> FileToString(const std::filesystem::path& file_name) {
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:181]: ReadFileToString

File: OrbitService/Utils.cpp [hunk 5]
```diff
@@ -18,54 +21,60 @@
...
 }
+}  // namespace
 
 ErrorMessageOr<std::vector<ModuleInfo>> ListModules(int32_t pid) {
+  std::filesystem::path proc_maps_path{absl::StrFormat("/proc/%d/maps", pid)};
+  OUTCOME_TRY(proc_maps_data, FileToString(proc_maps_path));
+  return ListModules(proc_maps_data);
+}
+
+ErrorMessageOr<std::vector<ModuleInfo>> ListModules(std::string_view proc_maps_data) {
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:67]: ParseMaps?

File: OrbitService/Utils.cpp [hunk 6]
```diff
@@ -18,54 +21,60 @@
...
+        absl::StrFormat("Failed to execute command \"%s\": %s", cmd, SafeStrerror(errno)));
   }
   while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
     result += buffer.data();
   }
   return outcome::success(result);
 }
+}  // namespace
 
 ErrorMessageOr<std::vector<ModuleInfo>> ListModules(int32_t pid) {
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:61]: ReadModules maybe? (not sure about this.)

File: OrbitService/UtilsTest.cpp [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+  {
+    // Example data
+    const std::string data{absl::StrFormat(
+        "7f6874285000-7f6874288000 r--p 00000000 fe:01 661216                     %s\n"
+        "7f6874288000-7f687428c000 r-xp 00003000 fe:01 661216                     %s\n"
+        "7f687428c000-7f687428e000 r--p 00007000 fe:01 661216                     %s\n"
+        "7f687428e000-7f687428f000 r--p 00008000 fe:01 661216                     %s\n"
+        "7f687428f000-7f6874290000 rw-p 00009000 fe:01 661216                     %s\n",
+        hello_world_path, hello_world_path, hello_world_path, hello_world_path, hello_world_path)};
```
Comment (dimitry-) [inline OrbitService/UtilsTest.cpp:57]: maybe test for 2 modules here?

File: OrbitService/UtilsTest.cpp [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    ASSERT_TRUE(result) << result.error().message();
+    EXPECT_TRUE(result.value());
+  }
+
+  {
+    const std::string data =
+        "/proc/1507305/exe: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically "
+        "linked, interpreter /lib64/ld-linux-x86-64.so.2, "
+        "BuildID[sha1]=16e95e7df322f1e9a8fb395dec25edc55e61fae7, for GNU/Linux 3.2.0, with "
+        "debug_info, not stripped";
```
Comment (dimitry-) [inline OrbitService/UtilsTest.cpp:124]: that test can go away if we use ElfFile.

File: OrbitService/Utils.cpp (hunk 3 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp]: good idea, done

File: OrbitService/Utils.cpp (hunk 4 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp]: done

File: OrbitService/Utils.cpp (hunk 5 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp]: done

File: OrbitService/Utils.cpp (hunk 6 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp]: Yes, it does not matter that much. I like ReadModules -> done

File: OrbitService/UtilsTest.cpp (hunk 7 above)
Comment (antonrohr) [reply OrbitService/UtilsTest.cpp]: done

File: OrbitService/UtilsTest.cpp (hunk 8 above)
Comment (antonrohr) [reply OrbitService/UtilsTest.cpp]: done

File: OrbitService/Utils.cpp [hunk 9]
```diff
@@ -121,32 +130,32 @@ ErrorMessageOr<std::vector<ModuleInfo>> ListModules(int32_t pid) {
   return result;
 }
 
-outcome::result<std::unordered_map<pid_t, double>> GetCpuUtilization() {
-  const std::string cmd = "top -b -n 1 | sed -n '8, 1000{s/^ *//;s/ *$//;s/  */,/gp;};1000q'";
-  OUTCOME_TRY(result, ExecuteCommand(cmd));
+ErrorMessageOr<std::unordered_map<pid_t, double>> GetCpuUtilization() {
+  const std::string cmd = "top -b -n 1 -w512 | sed -n '8, 1000{s/^ *//;s/ *$//;s/  */,/gp;};1000q'";
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:134]: this is slow and prone to top output change. but I out of scope for this PR. I think this is last command we use right? nothing else uses command-line utils?

File: OrbitService/ProcessList.cpp [hunk 10]
```diff
@@ -78,17 +78,14 @@ ErrorMessageOr<void> ProcessList::Refresh() {
...
-
-    auto file_path_result = LinuxUtils::GetExecutablePath(pid);
+    auto file_path_result = utils::GetExecutablePath(pid);
     if (file_path_result) {
       process.set_full_path(std::move(file_path_result.value()));
+
+      const auto& elf_file = ElfUtils::ElfFile::Create(file_path_result.value().string());
+      if (elf_file) {
+        process.set_is_64_bit(elf_file.value()->Is64Bit());
+      }
```
Comment (dimitry-) [inline OrbitService/ProcessList.cpp:88]: Maybe log a warning for !elf_file?

File: OrbitService/Utils.cpp (hunk 9 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp:134]: Not anywhere else on the Service as far as I am aware. I would also put this in a different PR. The parameter I added (-w512) already solved a weird bug that is related to terminal window size, which I encountered coincidentally.

File: OrbitService/ProcessList.cpp (hunk 10 above)
Comment (antonrohr) [reply OrbitService/ProcessList.cpp:91]: done

**Comment by dimitry-** [comment]
Looping in @MariaSandru27

**Comment by antonrohr** [comment]
I was already planning on merging into master and locally already started rebasing to remove the fixups. But I am happy to incorporate more changes / ideas. @MariaSandru27 let me know when you find something or when you are done

**Pull request merged**